src/pkg/logger/logger.go

```diff
@@ -170,6 +170,30 @@ func WithContext(ctx context.Context, logger *slog.Logger) context.Context {
 	return context.WithValue(ctx, defaultCtxKey, logger)
 }
 
+// TODO (@austinabro321) once we switch over to the new logger completely the enabled key & logic should be deleted
+type ctxKeyEnabled struct{}
+
+var defaultCtxKeyEnabled = ctxKeyEnabled{}
+
+// WithLoggingEnabled allows stores a value to determine whether or not slog logging is enabled
+func WithLoggingEnabled(ctx context.Context, enabled bool) context.Context {
+	return context.WithValue(ctx, defaultCtxKeyEnabled, enabled)
+}
+
+// Enabled returns true if slog logging is enabled
+func Enabled(ctx context.Context) bool {
```

Comment:
There's nothing wrong with the other route of passing the flag in as config struct values, but this is consistent with the `From(ctx)` route we took with `*slog.Logger`. Better to stick to the one way to do this imo.